src/block/decompress_safe.rs

```diff
@@ -98,19 +99,18 @@ pub fn decompress_into_with_dict(
     output_pos: usize,
     ext_dict: &[u8],
 ) -> Result<usize, DecompressError> {
-    decompress_internal::<true>(input, &mut (output, output_pos).into(), ext_dict)
+    decompress_internal::<true>(input, &mut Sink::new(output, output_pos), ext_dict)
 }
 
 /// Decompress all bytes of `input` into `output`.
 ///
 /// Returns the number of bytes written (decompressed) into `output`.
-#[inline]
-fn decompress_internal<const USE_DICT: bool>(
+#[inline(always)]
```

Comment:
Are you sure about this? A inline that high can cause underlying inlines to be removed, because of some code size thresholds

Reply:
I need to re-test this. I'm 99% sure it's a leftover from some experimentation `#[inline]` -> `#[inline(always)]` here doesn't seem to affect performance w/ or w/o LTO.
Inlining on `compress_internal` causes a 10%+ of performance loss non-LTO benchmarks. Thus the `#[inline(never)]` annotation there.

Reply:
I checked this again and I remember why.. inline always adds 10% performance to `FrameDecompress/lz4_flex_rust_indep/34308` in _non-LTO_ builds. Not sure what black magic the compiler is doing.

Reply:
I should also say that this applies for the code in `master` as well, it's sort of unrelated to the PR title.